examples/char_count_zig/src/main.zig

```diff
@@ -24,3 +24,30 @@ fn char_count_zig(input_text: []const u8, target_char: []const u8) !u32 {
     }
     return count;
 }
+
+const Testsuite1 = struct {
+    pub fn testHappyPath() !void {
+        const input_text = "Hello World";
+        const target_char = "l";
+        const expected_count: u32 = 3;
+        const actual_count = try char_count_zig(input_text, target_char);
+        try std.testing.expectEqual(expected_count, actual_count);
+    }
+};
+
+const Testsuite2 = struct {
+    pub fn testMoreThanOneByteError() !void {
+        const input_text = "Hello World";
+        const target_char = "lo";
+        _ = char_count_zig(input_text, target_char) catch |err| {
+            try std.testing.expectEqual(error.PGErrorStack, err);
+            // TODO: how do we check the error message?
+            return;
+        };
+        try std.testing.expect(false);
+    }
+};
+
+comptime {
+    pgzx.testing.registerTests(.{ Testsuite1, Testsuite2 }, @import("build_options").testfn);
```

Comment:
I see, we moved the 'bool' back to the users code. Maybe we should swap the order of the argument then, so I can write: 

```
comptime {
  pgzx.testing.registerTests(@import("build_options").testfn, .{
    TestSuite1,
    TestSuite2,
  });
}
```